soc/espressif/soc.yml

```diff
@@ -0,0 +1,15 @@
+family:
+  - name: espressif_esp32
+    series:
+      - name: esp32
+        socs:
+          - name: esp32
+      - name: esp32s2
+        socs:
+          - name: esp32s2
+      - name: esp32s3
+        socs:
+          - name: esp32s3
+      - name: esp32c3
+        socs:
+          - name: esp32c3
```

Comment:
this is not correct, one or more of these have multiple CPUs, so why are they not listed?

Reply:
I think you are referring to clustering options, right? (appcpu and procpu instead)

Reply:
Yes, 2 CPUs or core means it needs to have a `cpucluster`

Reply:
@nordicjm Having that split with cpucluster requires more modifications, not only regarding V2 model porting. Keeping that as it has been, i.e, different type of boards, allows AMP/IPM stuff to work: `samples/drivers/ipm/ipm_esp32/`. I suggest we keep this way and create a new PR (which I will) to improve AMP support and allow proper cpu clustering.

Reply:
Why not acceptable? What is the issue on having 2 different board variant within the same board? Is having more that 1 board variant in the same board spec wrong?

Reply:
I'll add those cpuclusterings and evaluate the whole effort to make this work. I'll ping back soon.